packages/pds/src/api/com/atproto/session.ts

```diff
@@ -37,14 +34,97 @@ export default function (server: Server) {
       )
     }
 
-    const jwt = auth.createToken(user.did)
+    const access = auth.createAccessToken(user.did)
+    const refresh = auth.createRefreshToken(user.did)
+    await db.db
+      .insertInto('refresh_token')
+      .values({
+        id: refresh.payload.jti,
+        did: refresh.payload.sub,
+        expiresAt: new Date(refresh.payload.exp * 1000).toISOString(),
+      })
+      .execute()
+
+    return {
+      encoding: 'application/json',
+      body: {
+        did: user.did,
+        name: user.username,
+        accessJwt: access.jwt,
+        refreshJwt: refresh.jwt,
+      },
+    }
+  })
+
+  server.com.atproto.refreshSession(async (_params, _input, req, res) => {
+    const { db, auth } = locals.get(res)
+    const did = auth.getUserDidOrThrow(req, AuthScopes.Refresh)
+    const user = await db.getUser(did)
+    if (!user) {
+      throw new InvalidRequestError(
+        `Could not find user info for account: ${did}`,
+      )
+    }
+
+    const lastRefreshId = auth.verifyToken(auth.getToken(req) ?? '').jti
+    if (!lastRefreshId) {
+      throw new Error('Unexpected missing refresh token id')
+    }
+
+    const access = auth.createAccessToken(user.did)
+    const refresh = auth.createRefreshToken(user.did)
+    await db.transaction(async (dbTxn) => {
+      const { numDeletedRows } = await dbTxn.db
+        .deleteFrom('refresh_token')
+        .where('id', '=', lastRefreshId)
+        .executeTakeFirst()
+      if (numDeletedRows < 1) {
+        throw new InvalidRequestError('Token has been revoked', 'ExpiredToken')
+      }
+      await dbTxn.db
+        .insertInto('refresh_token')
+        .values({
+          id: refresh.payload.jti,
+          did: refresh.payload.sub,
+          expiresAt: new Date(refresh.payload.exp * 1000).toISOString(),
+        })
+        .execute()
+    })
+
     return {
       encoding: 'application/json',
-      body: { jwt, name: user.username, did: user.did },
+      body: {
+        did: user.did,
+        name: user.username,
+        accessJwt: access.jwt,
+        refreshJwt: refresh.jwt,
+      },
     }
   })
 
-  server.com.atproto.deleteSession(() => {
-    throw new InvalidRequestError('Not implemented')
+  server.com.atproto.deleteSession(async (_params, _input, req, res) => {
+    const { db, auth } = locals.get(res)
+    const token = auth.getToken(req)
+    if (!token) {
+      throw new AuthRequiredError()
+    }
+    const lastRefreshId = auth.verifyToken(token, AuthScopes.Refresh).jti
+    if (!lastRefreshId) {
+      throw new Error('Unexpected missing refresh token id')
+    }
+
+    // If the token was already revoked, that's alright.
```

Comment:
if it's alright if the token is already revoked, then should we be throwing if not found?

Reply:
seems fine to make this idempotent & if the session is already deleted, just returning 200

Reply:
In the latest commit I removed that erroneous comment (good catch!)  It was there because I originally wrote it to be idempotent, but then balked once I realized that I was already throwing during token verification when the token was expired.  I figured either both cases should throw, or both cases should not throw.

But I like that a lot more— I will take a pass at this to ignore both token expiration and revocation 👍

Reply:
makes sense 👍